File: app/src/main/java/com/github/goldy1992/mp3player/service/library/search/SearchDao.java
package com.github.goldy1992.mp3player.service.library.search;

import java.util.List;

public interface SearchDao<T extends SearchEntity> {

    void insert(T t);

    int getCount();

    void insertAll(List<T> songs);

    List<T> query(String query);
}

File: service/src/main/java/com/github/goldy1992/mp3player/service/library/search/SearchDao.kt
package com.github.goldy1992.mp3player.service.library.search

interface SearchDao<T : SearchEntity> {


    fun insert(t: T)
    fun count() : Int
    fun insertAll(items: List<@JvmSuppressWildcards T>)
    fun deleteOld(ids: List<String?>?)
    fun query(query: String?): List<T>?
}